src/Lower.cpp

```diff
@@ -236,7 +236,7 @@ void lower_impl(const vector<Function> &output_funcs,
     log("Lowering after uniquifying variable names:", s);
 
     debug(1) << "Simplifying...\n";
-    s = simplify(s, false);  // Storage folding and allocation bounds inference needs .loop_max symbols
+    s = simplify(s);
```

Comment:
Seems like a great opportunity to simplify the simplifier! (In a follow up PR of course)